### PR TITLE
Cleanup frontend

### PR DESCRIPTION
**Pull request by Trinovantes**
A lot of small changes that I found while going through the source code.

- Removed coffeescript since we're already mostly JS
- Use ES6 to clean up some code (e.g. default function param)
- Setup browserify in gulp task "the right way". This includes compiling
  Handlebars templates with `hbsfy`. This also allows us to extend our test
  coverage to views since the `templates` attribute can now be handled with
  dependency injection.
- Localized gulp packages being loaded to avoid polluting the global
  namespace in `gulpfile.js`
- Updated logger initialization so that it's cleaner
- Updated `API_BASE` in Config file to include `/api`
- Moved `DEBUG` flag to Config file
- Added empty files for builder

### CODE REVIEW

File: src-backbone/app/js/templates/procedures/builder_layout.hbs [hunk 1]
```diff
@@ -0,0 +1 @@
+hello world!
```
Comment (Ommy) [inline src-backbone/app/js/templates/procedures/builder_layout.hbs]: Required?

**Comment by Ommy** [comment]
RIP in F CoffeeScript :cry: 

LGTM after comment

File: src-backbone/app/js/views/procedures/procedures_layout.js [hunk 2]
```diff
@@ -78,7 +78,9 @@ module.exports = Marionette.LayoutView.extend({
         event.preventDefault();
 
         var self = this;
-        var new_procedure = new this.Procedure({}, {
+        var new_procedure = new this.Procedure({
+            author: 'Stephen' // TODO either fetch user's username or let API allow nullable authors
```
Comment (ConnorCimowsky) [inline src-backbone/app/js/views/procedures/procedures_layout.js]: just making sure this is intentional

**Comment by ConnorCimowsky** [comment]
rest in pepperoni, coffeescript

**Comment by ConnorCimowsky** [comment]
LGTM :sparkles:

File: src-backbone/app/js/views/procedures/procedures_layout.js (hunk 2 above)
Comment (Trinovantes) [inline src-backbone/app/js/views/procedures/procedures_layout.js]: Yea this is actually the correct way to initialize a model. I was actually surprised the previous way worked too

**Pull request merged**